stage1/enter/enter.c

```diff
@@ -27,6 +27,12 @@
 #include <sys/wait.h>
 #include <unistd.h>
 
+#ifdef USE_NR_SETNS_MACRO_INSTEAD
```

Comment:
That won't work - preprocessor does not tell you if the function is defined or not.

What I could do is to use your code, but replace the `#ifndef setns` with `#USE_NR_SETNS_MACRO_INSTEAD`. This macro is defined only when we don't have setns function.